tests/pytests/test_vecsim.py

```diff
@@ -930,4 +930,77 @@ def test_fail_on_v1_dialect():
                         'DIM', dim, 'DISTANCE_METRIC', 'COSINE')
     conn.execute_command("HSET", "i", "v", one_vector.tobytes())
     res = env.expect("FT.SEARCH", "idx", "*=>[KNN 10 @v $BLOB]", "PARAMS", 2, "BLOB", one_vector.tobytes())
-    res.error().contains("Syntax error")
\ No newline at end of file
+    res.error().contains("Syntax error")
+
+
+def test_hybrid_query_with_global_filters():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+    dim = 2
+    index_size = 1000 * env.shardsCount
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'v', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32',
+               'DIM', dim, 'DISTANCE_METRIC', 'L2', 't', 'TEXT', 'num', 'NUMERIC', 'coordinate', 'GEO').ok()
+
+    p = conn.pipeline(transaction=False)
+    for i in range(1, index_size+1):
+        vector = np.full(dim, i, dtype='float32')
+        conn.execute_command('HSET', i, 'v', vector.tobytes(), 't', 'hybrid', 'num', i/100, 'coordinate',
+                             str(i/100)+","+str(i/100))
+    p.execute()
+    env.assertEqual(to_dict(env.cmd("FT.DEBUG", "VECSIM_INFO", 'idx', 'v'))["INDEX_SIZE"], index_size)
+    query_data = np.full(dim, index_size, dtype='float32')
+
+    # Run VecSim query in KNN mode (non-hybrid), and expect to find only one result (with key=index_size-2).
+    inkeys = [index_size-2]
+    expected_res = [1, str(index_size-2), ['__v_score', str(dim*2**2)]]
+    env.expect('FT.SEARCH', 'idx', '*=>[KNN 10 @v $vec_param]', 'INKEYS', len(inkeys), *inkeys,
+               'RETURN', 1, '__v_score', 'PARAMS', 2, 'vec_param', query_data.tobytes()).equal(expected_res)
+
+    # Change the text value to 'other' for 20% of the vectors (with ids 5, 10, ..., index_size)
+    for i in range(1, int(index_size/5) + 1):
+        vector = np.full(dim, 5*i, dtype='float32')
+        conn.execute_command('HSET', 5*i, 'v', vector.tobytes(), 't', 'other')
+
+    # Run VecSim query in hybrid mode, expect to get only the vectors that passes the filters
+    # (index_size-10 and index_size-100, since they has "other" in its 't' field and its id is in inkeys list).
+    inkeys = [index_size-2, index_size-10, index_size-100]
+    expected_res = [2, str(index_size-100), ['__v_score', str(dim*100**2)], str(index_size-10), ['__v_score', str(dim*10**2)]]
+    env.expect('FT.SEARCH', 'idx', '(other)=>[KNN 10 @v $vec_param]', 'INKEYS', len(inkeys), *inkeys,
+               'RETURN', 1, '__v_score', 'PARAMS', 2, 'vec_param', query_data.tobytes()).equal(expected_res)
+
+    # Test legacy numeric and geo global filters
+    expected_res = [10]
+    # Expect to get top 10 ids where maximum is index_size/2 in the index (due to the numeric filter).
+    for i in range(10):
+        expected_res.append(str(int(index_size/2-i)))
+        expected_res.append(['__v_score', str(int(dim*(index_size/2+i)**2))])
+    env.expect('FT.SEARCH', 'idx', '*=>[KNN 10 @v $vec_param]', 'filter', 'num', 0, index_size/200, 'SORTBY', '__v_score',
```

Comment:
`index…size/200` can you explain?

Reply:
I changed it, now the `num` field equals to the doc id